### ISSUE TITLE
Transect walk data entry - support for species names in other (non-Latin) languages

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/ABLE#507]
https://butterfly-monitoring.net/walk-data-entry

Currently not able to enter data for species using languages other than Latin and English?

Need to be able to enter names in Japanese for example.  Names in other languages are stored as synonyms currently

**Comment by DavidRoy** [comment]
@andrewvanbreda any progress on this one?

**Comment by andrewvanbreda** [comment]
@DavidRoy 
I spent a few hours on this today. 
I reproduced the problem on my machine, and I can the see the rough kind of thing than needs to be done to fix it. However I have not got it actually working yet as it is doing something a bit odd.
However, what I am adding is similar to other pages, so it is likely I am just missing something I haven't noticed yet.I am confident with a little more effort it will suddenly start working.
(It is one of those ones that is either going to be not working, or working perfectly, there is no middle ground)

**Comment by JimBacon** [comment]
Is https://github.com/BiologicalRecordsCentre/ABLE/issues/472 relevant?

In particular, the para which says 

> When selecting species, the common names that are offered depend upon the user language setting. Since the language switcher is the more visible way to set the language the user setting may stay on the default of English and so the recorder is only offered English common names.

**Comment by andrewvanbreda** [comment]
@JimBacon 
Essentially the problem I am experiencing is the EBMS forms do not include the standard dynamic.php code for translations on the sample termlists.
I have corrected on my machine and about to deploy to my avb-develop multisite branch for further testing.
This is different to the link you have included above, however referencing that issue here is not going to do any harm, it might include some notes that might be useful for testing.

**Comment by JimBacon** [comment]
Translation of termlists sounds like a different (though important) issue from the one David is reporting above, unless I have misunderstood.

**Comment by andrewvanbreda** [comment]
@JimBacon 
Thanks for pointing this out.
This confusion is because I had been talking to David over email about Japanese not working, and in those cases we had be talking about location editing page attributes, and in that example what we had been working on was actually a location attribute.
The problem is likely universal and similar. I will check I get the species names also are working.

**Comment by JimBacon** [comment]
I have a pretty clear memory that species names do work, provided the language is set in the user profile.

**Comment by andrewvanbreda** [comment]
@JimBacon OK, thanks for the tip. I will look at the whole language thing.

**Comment by andrewvanbreda** [comment]
Hi @DavidRoy @JimBacon,

I have examined the data entry page and the code.

The following notes are in reference to the species list language problem only (these observations do not apply to the termlist language issue)

Currently the page is configured in background to show the "Preferred Taxon (usually latin)", so the lists are going to show the latin regardless of the language.
That is obviously the case even if all the language settings are set to Japanese (both account page and site language real-time switcher), but I tried it anyway.

So the obvious thing to do is switch the taxon name configuration option to "Common".
However this seems to always show English.
This again shows English regardless of any language selections.

I examined the code, and I think the language is not even passed to the Warehouse in the code.

I also notice that a new version 2 of the data entry page is in development.
I know that is definitely the case as I now remember Gary mentioning it.

So I will phone Gary to find out what to expect from the new page, and whether the code needs altering in the old page.
It would also be useful for me to get a proper overview from him about what he expects to be working on the site multi-lingual wise.
I am slightly confused why my conclusions are not consistent with previous observations made in the other thread, although I agree with the previous conclusions about the language selections being confusing.

**Comment by DavidRoy** [comment]
Thanks @andrewvanbreda.
Just a thought but I wonder if the walk data entry page should give an option to use the language for the species lists, regardless of other language settings?

Let me know the outcome of discussions with Gary

**Comment by andrewvanbreda** [comment]
Hi @DavidRoy @JimBacon,

I have just spoken to Gary.

To put simply, he checked the code and the species list are not going to work with languages at the moment, even if the common name option is used.
He will look at this tomorrow himself.

I did ask him what he expected the rest of the EBMS system to be capable of in regards to translation, and he said it should work with Drupal translate, but I need to test this further.
I have already done some work regarding this, and my tests show we do need to use Drupal translate for termlists, because if we use Warehouse translate, when we go back to edit the entered data, the Indicia page will appear blank if we view it with a different language to the one that was originally used for data entry.
This is a problem we had with UKPoMS.
To get around this, what we do is tell Drupal to do the translation.
So I will test this further.
(We can still add Warehouse translations in addition to this for reporting purposes).

In terms of the other suggestion of displaying the species list in the language for the species list regardless of setting. 
I suppose we could perhaps fake this type of behaviour, and hold a preferred language on the page configuration itself and then pass this to the Warehouse, but perhaps we best worry about this after the other issues. I didn't discuss this with Gary as only just thought of it.

Anyway, in summary me and Gary have more work to do, and will get back to you.

Andy

**Comment by andrewvanbreda** [comment]
Hi @Gary-van-Breda 
Ok thanks for the info.
I think I can fix the termlists, leave that wth me as I have some other code that fixes them for the edit transect page I haven't deployed yet. It should be the same thing.
I will look at the species list bit, but can't comment at moment untill I have looked and got my head around it.

**Comment by andrewvanbreda** [comment]
@JimBacon 
Thanks for the info, actually that is a box I couldn't get to work, which is something Gary was looking at.

**Comment by andrewvanbreda** [comment]
General note : I have deployed the fix for the termlists to the ebms-site-details page, and sent David an email with instructions.
I will not put any further details about this here as it moves away from the original intent of this particular issue. However I will note that standard Indicia pages should not have any trouble, but EBMS specific pages will need a fix for termlists.

**Comment by Gary-van-Breda** [comment]
@andrewvanbreda : I've pushed a change for this issue onto the master branch. Do you want to have a check?

**Comment by andrewvanbreda** [comment]
@Gary-van-Breda 
Yes will take a look when get chance. Thanks for the changes.

**Comment by andrewvanbreda** [comment]
Hi @Gary-van-Breda 
I deployed this to TEST, as on DEV the Walk Data Entry form is the version 1 of the form.

I have tried, but I can't get the Display Taxon Field option to work whatever I try (including setting language in the switcher, or on the account page, or both).
Even if set to common, it always shows everything in latin.
Any thoughts on this? Have I got the config wrong? Is there a bug you need to fix? (or do you have a rough idea what the issue might be to save me time if you want me to fix it).

Hi @DavidRoy
The species list has a Preferred option, or a Common Name option, do we need an extra option that is a mixture? At the moment if you switch on the language specific names, we'll loose the latin name.

**Comment by Gary-van-Breda** [comment]
@andrewvanbreda @DavidRoy : Apologies - I'd missed the "Preload species list" change event: now fixed and pushed through to Test.

Just to confirm for the rows in the species grids the new functionality is:

- if set to "preferred taxon", the name is shown in latin, with the title (i.e. hover text on the name) as the common name in the requested language, or if no common name exists in that language, then the hover text is the default common name (if any exists).
- If set to "common", the name is shown as the common name in the requested language, with the title (i.e. hover text on the name) as the latin, or if no common name exists in that language, then the name is shown in latin with the hover text as the default common name.

This use of the hover text to show either the latin name or the common name means that we do not need an extra option.

**Comment by andrewvanbreda** [comment]
Hi @Gary-van-Breda 
Thanks the info about the modes, yes that covers the need for the mix of names.

I am still having difficulty getting this to work though.

The first trouble I have is no matter what I do, the list is always displayed as a latin name with English hint.
Not matter the language switcher setting, no matter the site language settings, also changing the Display Taxon field setting makes no difference.
See screenshot (The species showing the hint in the shot is the example one we spoke about which has many languages).
I tried German as well as Japanese with no luck

I still can't get the search box to work either. Whenever I click on this it seems to show a grey No Entry style icon and doesn't focus. There doesn't seem to be anything much reported by Firefox developer tools apart from one red warning favicon.ico not found. Also some general warning about the site, but I don't think those would affect it.
I am less worried about this as no-one else seems to be reporting it, and it has always done it for me. Although I would be curious why, as it does prevent me testing it.

![EBMS screenshot](https://user-images.githubusercontent.com/2463614/214280956-81043f00-9d16-47d1-a4ce-62ecc8da9b8c.jpg)

**Comment by JimBacon** [comment]
I have a feeling the search box is disabled until all the sun, cloud and reliability fields have been completed.

**Comment by andrewvanbreda** [comment]
@DavidRoy @Gary-van-Breda @JimBacon 
I have checked this and deployed to live.
I spoke to Gary yesterday, and he found the problem was that the code hadn't been deployed to test, but actually it does seem to work fine.
David, close if happy.

**Comment by DavidRoy** [comment]
@andrewvanbreda I've tested and when masquerading as user assigned to 'Japan BMS' that the text is in Japanese.

The species list is in Latin and the hover-over shows the Japanese common name.

Is it possible to show the common names in Japanese in the data grid?

**Comment by andrewvanbreda** [comment]
Hi @DavidRoy 
Do you want to try this on the test site first. 

https://test-brc-ebms.pantheonsite.io/walk-data-entry

I have switched on the Common Name option for the grid.

I think the potential difficulty here is that not all the names in the list will have translations, so you end up with a mixture.
If you prefer this I can switch Live to be like this, if you want it to work differently, I may or may not need to ask Gary.

**Comment by DavidRoy** [comment]
@andrewvanbreda this works well and can be moved to live.  As you say, there might still need to be some translation work for the species list

**Comment by CrisSevilleja** [comment]
It is working well @andrewvanbreda thank you for this work. 
Ready to close

**Comment by andrewvanbreda** [comment]
Hi @CrisSevilleja
Glad working well.
I can't take the credit though, it was really @Gary-van-Breda who did the code on the species list.

**Comment by andrewvanbreda** [comment]
Closing as sounds like everyone is happy.
